app/controllers/providers/confirm_dwp_non_passported_applications_controller.rb

```diff
@@ -5,6 +5,7 @@ class ConfirmDWPNonPassportedApplicationsController < ProviderBaseController
 
     def show
       delete_check_benefits_from_history
+      remove_dwp_override
```

Comment:
Or, from the CYA page, they visit this and hit the back button.  As a user I'd expect my original data to be present but this would remove it.

Something like 
```ruby
    def update
      return continue_or_draft if draft_selected?

      @form = Providers::ConfirmDWPNonPassportedApplicationsForm.new(form_params)

      if @form.valid?
        remove_dwp_override if correct_dwp_result? # <-- pseudo code 
        update_joint_benefit_response
        update_application_state
        HMRC::CreateResponsesService.call(legal_aid_application) if make_hmrc_call?
        return go_forward(correct_dwp_result?)
      end
      render :show
    end
```